x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -484,6 +485,12 @@ private void consumeUser(User user, Map<Realm, Tuple<String, Exception>> message
                     final String cause = tuple.v2() == null ? "" : " (Caused by " + tuple.v2() + ")";
                     logger.warn("Authentication to realm {} failed - {}{}", realm.name(), message, cause);
                 });
+                List<Realm> unlicensedRealms = realms.getUnlicensedRealms();
+                if (unlicensedRealms.isEmpty() == false) {
+                    logger.warn("Authentication failed." +
```

Comment:
Maybe we can split this to two warnings, one saying that authentication failed because user could not be authenticated to any of the available realms (which we can log either way) and another saying that these X realms were skipped even though configured as they are not permitted in the current license ?

Reply:
I kept it in 1 message because experience says that users don't read more than 1 line. So if we split it, then they'll read the "Authentication failed" line, but then ignore the line under it that provides the useful info.

Really that `Authentication failed.` bit is just there to say **If you are trying to debug an authentication problem then this info is actually relvant to you, please don't ignore it.**

Reply:
Yes, I see your point. In my suggestion, it would make sense to me to add the first of the messages even outside the context of this PR - maybe this is an indication that it should not be added here then :)

My concern was that the way it is worded now, it sort of implies that the authentication failed _because_ there were skipped realms, which might or might not be the case. I know you explicitly call out the fact that this is not the case in the description of the PR, but I think reading the log message, that implication is there. 

I'd prefer if we remove the "Authentication failed ." part or if we added a second message, or make it less implied in the existing one *but* I don't feel strongly about it

Reply:
> it would make sense to me to add the first of the messages even outside the context of this PR

We've had feedback before about logging too much when auth fails. I don't think users want us to have a `WARN` or even `INFO` message when there's a standard auth failure.
I'd be happy with a `DEBUG`, but I don't want to do it here.

> My concern was that the way it is worded now, it sort of implies that the authentication failed because there were skipped realms, which might or might not be the case

OK, let's fix that, because we do what to avoid that impression if we can.

Reply:
What if we changed it to 
```
Authentication failed using realms [native/native1, file/file1]. Realms [ldap/ldap1,pki/pki1] were skipped because they are not permitted on the current license
```